File: cpp/pangram/pangram.cpp
#include "pangram.h"

#include <iostream>

namespace pangram {

bool is_pangram(std::string input) {
    const size_t letter_count = 26;

    if (letter_count > input.size()) {
        return false;
    }

    std::string text{};

    std::transform(input.begin(), input.end(), std::back_inserter(text),
                   tolower);

    std::map<char, bool> tracker{};

    for (auto rune : text) {
        if (!(bool)isalpha(rune)) {
            continue;
        }

        tracker[rune] = true;
    }

    const size_t count{tracker.size()};

    return letter_count == count;
}

} // namespace pangram

File: dart/pangram/lib/pangram.dart
class Pangram {
  bool isPangram(String text) {
    if (text.isEmpty | (text.length < 26)) {
      return false;
    }

    var letters = <String>{};
    letters = text
        .toLowerCase()
        .split('')
        .where((rune) => RegExp(r'^[a-z]$').hasMatch(rune))
        .toSet();

    return letters.length == 26;
  }
}
